pkg/cache/cache.go

```diff
@@ -552,8 +552,9 @@ func (c *Cache) AddClusterQueue(ctx context.Context, cq *kueue.ClusterQueue) err
 			continue
 		}
 		c.addOrUpdateWorkload(&workloads.Items[i])
-		if _, ok := cqImpl.admittedWorkloadsPerQueue[w.Spec.QueueName]; ok {
-			cqImpl.admittedWorkloadsPerQueue[w.Spec.QueueName]++
+		qKey := workload.QueueKey(&w)
+		if _, ok := cqImpl.admittedWorkloadsPerQueue[qKey]; ok {
```

Comment:
oops, can you add a unit test for this?

Reply:
Probably, we don't need to add a unit test anymore.